konan_sdk/konan_service/services.py

```diff
@@ -0,0 +1,78 @@
+from typing import List, Type
+
+from fastapi import FastAPI, Response
+from fastapi_utils.cbv import cbv
+
+from konan_sdk.konan_service.routers import KonanServiceRouter
+from konan_sdk.konan_service.serializers import (
+    KonanServiceBaseFeedback, KonanServiceBaseEvaluateRequest, KonanServiceBaseEvaluateResponse
+)
+
+
+class KonanService():
+    """Class that implements a Konan webservice
+    """
+    def __init__(
+        self,
+        predict_request_class: Type,
+        predict_response_class: Type,
+        model_class: Type,
+        feedback_target_class: Type = None,
+        evaluate_response_class: Type = KonanServiceBaseEvaluateResponse,
+        *model_args, **model_kwargs,
+    ) -> None:
+        """Initializes a konan service
+
+        Args:
+            predict_request_class (Type): Type of a prediction request.
+                Should be a class that inherits from KonanServiceBasePredictionRequest
+            predict_response_class (Type): Type of a prediction response.
+                Should be a class that inherits from KonanServiceBasePredictionResponse
+            model_class (Type): Type of the model that does the prediction.
+                Should be a class that inherits from KonanServiceBaseModel.
+                Must implement the predict() and evaluate() methods
+            feedback_target_class (Type, optional): Type of feedback target.
+                Defaults to value of predict_response_class if None.
+            evaluate_response_class (Type, optional): Type of an evaluation response.
+                Should be a class that inherits from KonanServiceBaseEvaluateResponse.
+                Defaults to KonanServiceBaseEvaluateResponse.
+        """
+        self.model = model_class(*model_args, **model_kwargs)
+        self.app = FastAPI(openapi_url='/docs', docs_url='/swagger')
+        feedback_target_class = feedback_target_class or predict_response_class
+
+        class ServiceFeedback(KonanServiceBaseFeedback):
+            target: feedback_target_class
```

Comment:
I think we should override prediction attribute here too right? to utilize the passed custom class

Reply:
Yeah, that's correct .. fixed.